catalog.go

```diff
@@ -1,17 +1,19 @@
 // Copyright © 2022 Roberto Hidalgo <coredns-consul@un.rob.mx>
+// Contributions by Charles Powell, 2023
```

Comment:
@cbpowell hey, just had a look at your fork and we implemented aliases in a very similar manner, so I added a few lines here, hope that's ok with you.

I also had [a very similar thought](https://github.com/hashicorp/vault/pull/11084) re: your work around vault not being able to set meta tags, but approached it myself by pointing `{consul,vault,nomad}.{my.home.domain}` at my ingress gateway (i.e. the poorly named `service_proxy` key) which terminates `{my.home.domain}` TLS anyways.

Let me know if you have thoughts on this PR—which mostly is about adding aliases and wildcards because I want to play with [garagehq](https://garagehq.deuxfleurs.fr/), and about the project in general; your contributions would be very welcome too!

Reply:
Awesome, thanks! I’ll check it out - apologies for not submitting a PR myself, I was just branching it off for my own personal use at home…plus this was my first time writing in Go.

Reply:
No worries, I wrote this for use in my home too, so no pressure! I'll merge this soon along a few other changes I had in mind, but happy to hear your thoughts at any point in the future.